docs/patterns/cache-aside.md

```diff
@@ -1,156 +1,135 @@
+
+**Consistency**. Implementing the Cache-Aside pattern doesn't guarantee consistency between the data store and the cache. An item in the data store can be changed at any time by an external process, and this change might not be reflected in the cache until the next time the item is loaded. In a system that replicates data across data stores, this problem can become serious if synchronization occurs frequently.
+
+**Local (in-memory) caching**. A cache could be local to an application instance and stored in-memory. Cache-aside can be useful in this environment if an application repeatedly accesses the same data. However, a local cache is private and so different application instances could each have a copy of the same cached data. This data could quickly become inconsistent between caches, so it might be necessary to expire data held in a private cache and refresh it more frequently. In these scenarios, consider investigating the use of a shared or a distributed caching mechanism.
+
+## When to use this pattern
+
+Use this pattern when:
+
+- A cache doesn't provide native read-through and write-through operations.
+- Resource demand is unpredictable. This pattern enables applications to load data on demand. It makes no assumptions about which data an application will require in advance.
+
+This pattern might not be suitable:
+
+- When the cached data set is static. If the data will fit into the available cache space, prime the cache with the data on startup and apply a policy that prevents the data from expiring.
+- For caching session state information in a web application hosted in a web farm. In this environment, you should avoid introducing dependencies based on client-server affinity.
+
+## Example
+
+In Microsoft Azure you can use Azure Redis Cache to create a distributed cache that can be shared by multiple instances of an application. The `GetMyEntityAsync` method in the following code example shows an implementation of the Cache-Aside pattern based on Azure Redis Cache. This method retrieves an object from the cache using the read-though approach.
+
+An object is identified by using an integer ID as the key. The `GetMyEntityAsync` method generates a string value based on this key (the Azure Redis Cache API uses strings for key values) and tries to retrieve an item with this key from the cache. If a matching item is found, it's returned. If there's no match in the cache, the `GetMyEntityAsync` method retrieves the object from a data store, adds it to the cache, and then returns it. The code that actually retrieves the data from the data store has been omitted because it is data store dependent. Note that the cached item is configured to expire in order to prevent it from becoming stale if it's updated elsewhere.
+
+```csharp
+private static ConnectionMultiplexer Connection;

```

Comment:
Though the intent of this article is to demonstrate a caching pattern, I still would either point to or include initialization of ```ConnectionMultiplexer``` object. Especially Lazy Singleton initialization is recommended approach in this case, e.g.
```
        private static readonly Lazy<ConnectionMultiplexer> LazyConnection =
            new Lazy<ConnectionMultiplexer>(
                () => ConnectionMultiplexer.Connect(ConfigurationManager.AppSettings["MyRedisCacheConnectionString"]));

public static ConnectionMultiplexer Connection => LazyConnection.Value;
```
The above serves dual purposes:
1. Shows from where to obtain the connection information for Redis cache.
2. Shows a recommended approach towards initializing a cache connection.